gossip_compaction/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+//! Parquet compaction completion notifications over [gossip].
+//!
+//! This sub-system is composed of the following primary components:
+//!
+//! * [`gossip`] crate: provides the gossip transport, the [`GossipHandle`], and
+//!   the [`Dispatcher`]. This crate operates on raw bytes.
+//!
+//! * The outgoing [`CompactionEventTx`]: a schema-specific wrapper over the
+//!   underlying [`GossipHandle`]. This type translates the protobuf
+//!   [`CompactionEvent`] from the application layer into raw serialised bytes,
+//!   sending them over the underlying [`gossip`] impl.
+//!
+//! * The incoming [`CompactionEventRx`]: deserialises the incoming bytes from
+//!   the gossip [`Dispatcher`] into [`CompactionEvent`] and passes them off to
+//!   the [`CompactionEventHandler`] implementation for processing.
+//!
+//! Users of this crate should implement the [`CompactionEventHandler`] trait to
+//! receive change events, and push events into the [`CompactionEventTx`] to
+//! broadcast changes to peers.
+//!
+//! ```text
+//!         ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+//!                                                              │
+//!         │                    Application
+//!                                                              │
+//!         └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+//!                     │                           ▲
+//!                     │                           │
+//!                     │                           │
+//!                     │      CompactionEvent      │
+//!                     │                           │
+//!                     ▼                           │
+//!         ┌──────────────────────┐   ┌─────────────────────────┐
+//!         │  CompactionEventTx   │   │    CompactionEventRx    │
+//!         └──────────────────────┘   └─────────────────────────┘
+//!                     │                           ▲
+//!                     │                           │
+//!                     │       Encoded bytes       │
+//!                     │                           │
+//!                     │                           │
+//!        ┌ Gossip  ─ ─│─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─│─ ─ ─ ─ ─ ─ ─
+//!                     ▼                           │             │
+//!        │    ┌──────────────┐          ┌──────────────────┐
+//!             │ GossipHandle │          │    Dispatcher    │    │
+//!        │    └──────────────┘          └──────────────────┘
+//!                                                               │
+//!        └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+//! ```
+//!
+//! # Best Effort
+//!
+//! This underlying gossip subsystem is designed to provide best effort delivery
+//! of messages, and therefore best-effort delivery of parquet creation events,
```

Comment:
```suggestion
//! of messages, and therefore best-effort delivery of compaction events,
```

Technically correct as is, but as there's the new `ParquetFileEvent` might as well disambiguate 🤷